opaque_keys/__init__.py

```diff
@@ -352,44 +353,43 @@ def __deepcopy__(self, memo):
     def __setstate__(self, state_dict):
         # used by pickle to set fields on an unpickled object
         for key in state_dict:
-            if key in self.KEY_FIELDS:  # pylint: disable=no-member
+            if key in self.KEY_FIELDS:
                 setattr(self, key, state_dict[key])
-        self.deprecated = state_dict['deprecated']  # pylint: disable=assigning-non-slot
-        self._initialized = True  # pylint: disable=assigning-non-slot
+        self.deprecated = state_dict['deprecated']
+        self._initialized = True
 
     def __getstate__(self):
         # used by pickle to get fields on an unpickled object
         pickleable_dict = {}
-        for key in self.KEY_FIELDS:  # pylint: disable=no-member
+        for key in self.KEY_FIELDS:
             pickleable_dict[key] = getattr(self, key)
         pickleable_dict['deprecated'] = self.deprecated
         return pickleable_dict
 
     @property
-    def _key(self):
+    def _key(self) -> tuple:
```

Comment:
Is it worthwhile (or even possible) to include the subtypes of the tuple returned?

Reply:
It would be nice, but I don't think it's even possible, as the number of fields in the tuple depends on the `KEY_FIELDS` of the subclass. The type isn't `tuple[str, ...]` either, as some of the "key field" values are actually `ObjectId` type, not strings.